backends/platform/sdl/sailfish/sailfish.cpp

```diff
@@ -0,0 +1,144 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#define FORBIDDEN_SYMBOL_ALLOW_ALL
+
+#include "common/scummsys.h"
+#include "common/config-manager.h"
+#include "common/translation.h"
+
+#include "backends/platform/sdl/sailfish/sailfish.h"
+#include "backends/platform/sdl/sailfish/sailfish-window.h"
+
+#include "backends/fs/posix/posix-fs-factory.h"
+#include "backends/fs/posix/posix-fs.h"
+#include "backends/saves/default/default-saves.h"
+
+#include "backends/keymapper/action.h"
+#include "backends/keymapper/keymapper-defaults.h"
+#include "backends/keymapper/hardware-input.h"
+#include "backends/keymapper/keymap.h"
+#include "backends/keymapper/keymapper.h"
+
+#define ORG_NAME        "org.scummvm"
+#define APP_NAME        "scummvm"
+
+void OSystem_SDL_Sailfish::init() {
+	// Initialze File System Factory
+	_fsFactory = new POSIXFilesystemFactory();
+
+	_window = new SdlWindow_Sailfish();
+
+	// Invoke parent implementation of this method
+	OSystem_SDL::init();
+}
+
+void OSystem_SDL_Sailfish::initBackend() {
+	if (!ConfMan.hasKey("rotation_mode")) {
+		ConfMan.setPath("rotation_mode", "90");
+	}
+
+	if (!ConfMan.hasKey("savepath")) {
+		ConfMan.setPath("savepath", getDefaultSavePath());
+	}
+
+	// Create the savefile manager
+	if (_savefileManager == nullptr) {
+		_savefileManager = new DefaultSaveFileManager(getDefaultSavePath());
+	}
+
+	OSystem_SDL::initBackend();
+}
+
+Common::Path OSystem_SDL_Sailfish::getDefaultSavePath() {
+	Common::String saveRelPath;
+
+	const char *prefix = getenv("HOME");
+	if (prefix == nullptr) {
+		return Common::Path();
+	}
+
+	saveRelPath = ".local/share/" ORG_NAME "/" APP_NAME "/saves";
+
+	if (!Posix::assureDirectoryExists(saveRelPath, prefix)) {
+		return Common::Path();
+	}
+
+	Common::Path savePath(prefix);
+	savePath.joinInPlace(saveRelPath);
+
+	return savePath;
+}
+
+Common::Path OSystem_SDL_Sailfish::getDefaultConfigFileName() {
+	Common::String configPath;
+
+	const char *prefix = getenv("HOME");
+	if (prefix == nullptr) {
+		return Common::Path();
+	}
+
+	configPath = ".local/share/" ORG_NAME "/" APP_NAME ;
+
+	if (!Posix::assureDirectoryExists(configPath, prefix)) {
+		return Common::Path();
+	}
+
+	Common::Path configFile(prefix);
+	configFile.joinInPlace(configPath);
+	configFile.joinInPlace("scummvm.ini");
+
+	return configFile;
+}
+
+Common::Path OSystem_SDL_Sailfish::getDefaultLogFileName() {
+	Common::String logFile;
+
+	const char *prefix = getenv("HOME");
+	if (prefix == nullptr) {
+		return Common::Path();
+	}
+
+	logFile = ".cache/" ORG_NAME "/" APP_NAME "/logs";
+
+	if (!Posix::assureDirectoryExists(logFile, prefix)) {
+		return Common::Path();
+	}
+
+	Common::Path logPath(prefix);
+	logPath.joinInPlace(logFile);
+	logPath.joinInPlace("scummvm.log");
+
+	return logPath;
+}
+
+
+bool OSystem_SDL_Sailfish::hasFeature(Feature f) {
+	switch (f) {
+	case kFeatureFullscreenMode:
+	case kFeatureAspectRatioCorrection:
```

Comment:
Is it necessary to forcibly disable aspect ratio correction here?

Reply:
Nope. Came from some file I used as a template. Removed.